server/viame_utils/types/pipelines.py

```diff
@@ -0,0 +1,29 @@
+from typing import List, Optional
+
+from typing_extensions import TypedDict
```

Comment:
I definitely understand your concerns. I think I had created these typed dicts originally before we moved to pydantic, maybe not. Typed dicts are just dicts at runtime, so there's no serialization concern there. However, I'm open to changing these to pydantic models if there's demand for it.

Reply:
I'm less opposed to this because TypedDict has first-class language support and is basically just a dict.  Just wondered if you had an explicit reason to prefer this.  

Leave it for now, I think.